pkg/resources/endpoints.go

```diff
@@ -53,3 +35,66 @@ func ParentResourceFromService(name string) string {
 	}
 	return name[:li]
 }
+
+type ReadyPodCounter interface {
+	ReadyCount() (int, error)
+}
+
+type UpdateableReadyPodCounter interface {
+	ReadyPodCounter
+	UpdateName(string)
+}
```

Comment:
I feel like this is adding a lot of complexity and I don't immediately see that being needed. Could these interfaces be simplified to:

```golang
type ReadyPodCounter func(namespace, serviceName string) (int, error)
```

? It's kinda what we used to do in the activator (which we rolled back recently) but it'd allow you to easily mock that function without having to deal with the fake informers while making the default implementation kinda straightforward as well (it just closes over the informer).

WDYT?

Reply:
I think the second interface is redundant.At that point I am 50/50 between first interface and Markus's proposal (since at that point they're mostly equivalent, but I don't like the functor, since it becomes hard to capture state or requires one more level of wrapping).

Reply:
Recreating my thinking yesterday, it went something like:

1. It would be nice to fully capture these values and only have a simple `ReadyCount()` method with params.
1. It is nice, hooray.
1. Wait, when someone calls `Update()`, the captured service name is invalid, so I need to update that. I'll add `UpdateName()`.
1. It works, hooray.
1. Wait, now I have to create a non-scoped implementation for which `UpdateName()` is meaningless. Maybe split them apart?
1. They split apart OK, hooray.

So if I pop this stack a few times, the func type idea looks like a plausible alternative. I will kick it around today.

Reply:
Though if the service name is immutable (see discussion below), then my original assumption that I can capture the variables at creation is still sound. I'll play with that too.

Reply:
I decided not to go with the functor option, as `namespace` and `serviceName` are specific to one implementation.